bitnami/janusgraph/templates/deployment.yaml

```diff
@@ -0,0 +1,317 @@
+              # To be able to run with readOnlyRootFilesystem, we create the file and mount it as a volume
+              touch /base-dir/.gremlin_groovy_history
+              # Additionally copy the default content of the ext directory for console plugins installation
+              cp /opt/bitnami/janusgraph/ext/* /ext-dir
+          {{- if .Values.resources }}
+          resources: {{- toYaml .Values.resources | nindent 12 }}
+          {{- else if ne .Values.resourcesPreset "none" }}
+          resources: {{- include "common.resources.preset" (dict "type" .Values.resourcesPreset) | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: empty-dir
+              mountPath: /ext-dir
+              subPath: app-ext-dir
+            - name: empty-dir
+              mountPath: /base-dir
+              subPath: app-base-dir
+        {{- include "janusgraph.waitForStorage" . | nindent 8 }}
+        {{- if and .Values.volumePermissions.enabled .Values.persistence.enabled }}
+        # Adjust volume permissions
+        {{- include "janusgraph.volumePermissionsInitContainer" . | nindent 8 }}
+        {{- end }}
+        {{- if .Values.initContainers }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      containers:
+        - name: janusgraph
+          image: {{ include "janusgraph.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.containerSecurityContext "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.command "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: BITNAMI_DEBUG
+              value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+            - name: JANUSGRAPH_PORT_NUMBER
+              value: {{ .Values.containerPorts.gremlin | quote }}
+            - name: JAVA_OPTIONS
+              value: {{ include "janusgraph.javaOpts" . | quote }}
+            - name: JANUSGRAPH_JMX_METRICS_ENABLED
+              value: {{ ternary "true" "false" .Values.metrics.enabled | quote }}
```

Comment:
Should we set a `JANUSGRAPH_JMX_METRICS_PORT_NUMBER` env. variable (or similar) to ensure the chosen port for JMX metris is the one set at `.Values.containerPorts.jmx`?

Reply:
`JANUSGRAPH_JMX_METRICS_PORT_NUMBER` is not configured through an env var, but relies on `JAVA_OPTS`.

The `containerPorts.jmx` variable is used in the JavaOpts helper:
```
{{/*
Returns the Janusgraph JVM Java options for the conteiner
*/}}
{{- define "janusgraph.javaOpts" -}}
{{- if .Values.metrics.enabled -}}
{{ printf "%s -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=%d" .Values.javaOptions (int .Values.containerPorts.jmx) }}
{{- else -}}
{{ print .Values.javaOptions }}
{{- end -}}
{{- end -}}
```